parser/src/java/org/apache/hadoop/hive/ql/parse/HiveParser.g

```diff
@@ -1840,6 +1841,14 @@ tableImplBuckets
     -> ^(TOK_ALTERTABLE_BUCKETS $num)
     ;
 
+tableWriteOrdered
+@init { pushMsg("table sorted specification", state); }
+@after { popMsg(state); }
+    :
+      KW_WRITE KW_ORDERED KW_BY sortCols=columnNameOrderList
+    -> ^(TOK_ALTERTABLE_BUCKETS $sortCols?)
```

Comment:
we should support iceberg transform expressions as well, not just identity

Reply:
I will create a follow up ticket.